src/common/buffer.cc

```diff
@@ -1324,8 +1325,14 @@ static ceph::spinlock debug_lock;
     // make a new buffer.  fill out a complete page, factoring in the
     // raw_combined overhead.
     size_t need = round_up_to(len, sizeof(size_t)) + sizeof(raw_combined);
-    size_t alen = round_up_to(need, CEPH_BUFFER_ALLOC_UNIT) -
-      sizeof(raw_combined);
+    size_t alen = round_up_to(need, CEPH_BUFFER_ALLOC_UNIT_BASE);
+    if (_carriage == &_buffers.back()) {
+      size_t nlen = (_carriage->raw_length() + sizeof(raw_combined)) << 1;
```

Comment:
Hmm, `_carriage` doesn't necessarily need to be of the `raw_combined` type. How about rounding the `raw_length()` to `CEPH_BUFFER_ALLOC_UNIT_BASE`?

Reply:
hrm, we could, or just skip the sizeof calculations entirely and stick with base * powers-of-2

Reply:
This might work as well.

Reply:
@markhpc this comment still seems relevant

Reply:
yeah, I'm still jumping back and forth between master and this branch.  Sorry about the noise.